doc/Process_Console.md

```diff
@@ -0,0 +1,589 @@
+ - `Restarts`: How many times this process has crashed and been restarted by
+   the kernel.
+ - `Grants`: The number of grants that have been initialized for the process
+   out of the total number of grants defined by the kernel.
+ - `State`: The state the process is in.
+
+ Setup
+ -----
+
+ You need a device that provides the `hil::uart::UART` trait. This code
+ connects a `ProcessConsole` directly up to USART0:
+
+ ```rust
+ # use kernel::{capabilities, hil, static_init};
+ # use capsules::process_console::ProcessConsole;
+
+ pub struct Capability;
+ unsafe impl capabilities::ProcessManagementCapability for Capability {}
+
+ let pconsole = static_init!(
+     ProcessConsole<usart::USART>,
+     ProcessConsole::new(&usart::USART0,
+                  115200,
+                  &mut console::WRITE_BUF,
+                  &mut console::READ_BUF,
+                  &mut console::COMMAND_BUF,
+                  kernel,
+                  Capability));
+ hil::uart::UART::set_client(&usart::USART0, pconsole);
+
+ pconsole.start();
+ ```
+> Note: Using the process console might require allocating more stack to the kernel. This is done by modifying the `STACK_MEMORY` variable from the board's `main.rs`.
+
+ Using ProcessConsole
+ --------------------
+
+ With this capsule properly added to a board's `main.rs` and the Tock kernel
+ loaded to the board, make sure there is a serial connection to the board.
+ Likely, this just means connecting a USB cable from a computer to the board.
+ Next, establish a serial console connection to the board. An easy way to do
+ this is to run:
+
+ ```shell
+ $ tockloader listen
+[INFO   ] No device name specified. Using default name "tock".
+[INFO   ] No serial port with device name "tock" found.
+[INFO   ] Found 2 serial ports.
+Multiple serial port options found. Which would you like to use?
+[0]	/dev/ttyS4 - n/a
+[1]	/dev/ttyACM0 - "BBC micro:bit CMSIS-DAP" - mbed Serial Port
+
+Which option? [0] 1
+[INFO   ] Using "/dev/ttyACM0 - "BBC micro:bit CMSIS-DAP" - mbed Serial Port".
+[INFO   ] Listening for serial output.
+
+tock$ 
+ ```
+
+ For the examples below we will have 2 processes on the board: blink (which will `blink` all the LEDs that are 
+ connected to the kernel), and `c_hello` (which prints 'Hello World' when the console is started). Also, a micro:bit v2 board was used as support for the commands, so the results may vary on other devices.
+ We will assume that the user has a serial connection to the board, either by using tockloader or another serial port software.
+ With that console open, you can issue commands. For example, to get a list of the available commands, use the `help` command:
+ ```text
+tock$ help
+Welcome to the process console.
+Valid commands are: help status list stop start fault boot terminate process kernel panic
+ ```
+
+In the following paragraphs the outputs of the listed commands will be shown:
+
+ - To see all of the processes on the board, use `list`:
+
+ ```text
```

Comment:
All of these code blocks here down need to be indented so they render correctly as part of the list.